binding.cc

```diff
@@ -539,9 +545,16 @@ namespace zmq {
   Handle<Value> Socket::GetSockOpt(int option) {
     T value = 0;
     size_t len = sizeof(T);
-    if (zmq_getsockopt(socket_, option, &value, &len) < 0) {
-      NanThrowError(ExceptionFromError());
-      return NanUndefined();
+    while(true) {
+      int rc = zmq_getsockopt(socket_, option, &value, &len);
+      if ( rc < 0 && zmq_errno()==EINTR) {
+        continue;
+      } else if (rc<0) {
+        NanThrowError(ExceptionFromError());
+        return NanUndefined();
+      } else {
+        break;
+      }
```

Comment:
when the condition with `EINTR` is satisfied, here doesn't the `continue;` statement put us back to reiterate through the loop? when this is the case I'm concerned whether control ever reaches the end of that function